src/srs-filter.c

```diff
@@ -236,7 +280,9 @@ xxfi_srs_milter_envfrom(SMFICTX* ctx, char** argv) {
     syslog(LOG_DEBUG, "conn# %d[%i] - xxfi_srs_milter_envfrom(\"%s\")",
            cd->num, cd->state, argv[0]);
 
-  if (strlen(argv[0]) < 1 || strcmp(argv[0], "<>") == 0 || argv[0][0] != '<' || argv[0][strlen(argv[0])-1] != '>' || !strchr(argv[0], '@')) {
+  null_addr = (strcmp(argv[0], "<>") == 0);
```

Comment:
I appreciate breaking this down for readability. while you're at it, can you update to strncmp? (and likewise for strncmp in the next line?)